### PR TITLE
Add M306 documentation

### PR DESCRIPTION
**Pull request by CRCinAU**
Adds basic M306 documentation for PR https://github.com/MarlinFirmware/Marlin/pull/23751

### CODE REVIEW

**Comment by tombrazier** [comment]
Thank you.

**Comment by tombrazier** [comment]
@thinkyhead it would be handy to have this merged so I can edit it.

File: _gcode/M306.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+requires: MPCTEMP
+group: thermal
+
+codes: [ M306 ]
+
+notes:
+  - Requires `MPCTEMP`.
+  - View current values with [`M306`](/docs/gcode/M306.html) and no parameters.
+  - If `EEPROM_SETTINGS` is enabled, all hotend PID values are saved with [`M500`](/docs/gcode/M500.html), loaded with [`M501`](/docs/gcode/M501.html), and reset with [`M502`](/docs/gcode/M502.html).
```
Comment (thisiskeithb) [inline _gcode/M306.md:15]: > all hotend PID values are saved

Did you mean MPCTEMP values?

**Comment by thisiskeithb** [comment]
@tombrazier: You can add `https://github.com/CRCinAU/MarlinDocumentation` as a remote in the interim & send PRs to CRC's `master` branch & they will show up here once merged.

**Comment by tombrazier** [comment]
@CRCinAU can you add a note to say M306 T will move the hotend to 1mm above the bed and so the hotend should be checked for blobs before running it. And that the 1mm can be configured with MPC_TUNING_POS.

**Comment by CRCinAU** [comment]
> @CRCinAU can you add a note to say M306 T will move the hotend to 1mm above the bed and so the hotend should be checked for blobs before running it. And that the 1mm can be configured with MPC_TUNING_POS.

Done.

**Comment by CRCinAU** [comment]
@thinkyhead - Just wondering if this could be merged now?

**Pull request merged**